ci/check-protobuf-against-stable.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+readonly SRC_DIR="$(dirname "${SCRIPT_DIR}")"
+
+cd "${SRC_DIR}"
+readonly CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
```

Comment:
The `azure-pipelines.yml` file only runs on the `main` branch, so this seems unnecessary.

Reply:
Oh, I'd like to run this check for fix releases too. How can I do that?

Reply:
At the moment we don't have anything running after merge on release branches. What we've opted for instead is to enforce sequential merges on branches, so PR checks are what you'd want there I suppose. If you implement the next step of the design document (checking against target commit on PRs), I think that should cover it.

Bear in mind that release branches run their own version of the various YAML files, so if you want this to affect existing release branches you'll need to backport at least that bit.

Reply:
Thanks, I'll do so then. Also, the override mechanism will need to be disabled on release branches (but that makes sense regardless, I think).

Reply:
Done, ee20aaf